src/pages/StainingQC.tsx

```diff
@@ -42,24 +43,24 @@ export default function StainingQC({ info }: StainingQCProps) {
   const stanCore = useContext(StanCoreContext);
 
   const formMachine = React.useMemo(() => {
-    return createFormMachine<ResultRequest, RecordStainResultMutation>().withConfig({
-      services: {
-        submitForm: (ctx, e) => {
-          if (e.type !== 'SUBMIT_FORM') return Promise.reject();
+    return createFormMachine<ResultRequest, RecordStainResultMutation>().provide({
+      actors: {
+        submitForm: fromPromise(({ input }) => {
```

Comment:
I can see lot of these changes. I hope this is the new xstate way it should be used